wayland-logout.c

```diff
@@ -35,64 +35,84 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 
+// size of field sun_path in struct stsockaddr_un
+#define pathlen ((int)sizeof(((struct sockaddr_un *)NULL)->sun_path))
+
 int main()
 {
-    char socket_path[108];
+    char socket_path[pathlen];
     char *xdg_runtime_dir = getenv("XDG_RUNTIME_DIR");
     char *wayland_display = getenv("WAYLAND_DISPLAY");
 
-    if (!xdg_runtime_dir)
+    if (!xdg_runtime_dir && !wayland_display)
```

Comment:
I'd rather separate error handling messages for each environment variable.

Reply:
The idea is that if none of the variables are set, you can't continue. If `WAYLAND_DISPLAY` is set and starts with `/`, `XDG_RUNTIME_DIR` doesn't matter, otherwise `XDG_RUNTIME_DIR` also needs to be set. And finally `WAYLAND_DISPLAY` always needs to be set. The way these conditionals are laid out, it's easy to fail in the correct place with the relevant error message.

Reply:
I fail to see a problem with the following:
```
if (!wayland_display)
{
    err_msg();
    return -1;
}
if (!xdg_runtime_dir)
{
    err_msg();
    return -1;
}
```
.. continue since they are both set ..

Reply:
The idea is that `xdg_runtime_dir` doesn't need to be set, as long as `wayland_display` is an absolute path.

Reply:
Should be good now.